### PR TITLE
Include kubelet logs in output when running node e2e tests

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

File: test/e2e_node/runner/run_e2e.go [hunk 1]
```diff
@@ -96,11 +92,26 @@ func main() {
...
+		if tr.err != nil {
 			errCount++
+			glog.Infof("%s================================================================%s", blue, noColour)
+			glog.Infof("Failure Finished Host %s Test Suite %s %v", host, tr.testCombinedOutput, tr.err)
+			glog.V(2).Infof("----------------------------------------------------------------")
+			glog.V(5).Infof("Host %s Etcd Logs\n%s%s%s", host, gray, tr.etcdCominedOutput, noColour)
+			glog.V(5).Infof("----------------------------------------------------------------")
+			glog.V(5).Infof("Host %s Apiserver Logs\n%s%s%s", host, gray, tr.apiServerCominedOutput, noColour)
+			glog.V(5).Infof("----------------------------------------------------------------")
+			glog.V(2).Infof("Host %s Kubelet Logs\n%s%s%s", host, gray, tr.kubeletCominedOutput, noColour)
```
Comment (vishh) [inline test/e2e_node/runner/run_e2e.go]: We will need kernel logs as well - `/var/log/messages` at the least.

File: test/e2e_node/runner/run_e2e.go [hunk 2]
```diff
@@ -123,34 +134,56 @@ func WaitForUser() {
 	u.Done()
 }
 
-func runTests(fullhost string) ([]byte, error) {
+type TestResult struct {
+	fullhost               string
+	err                    error
+	testCombinedOutput     string
+	etcdCominedOutput      string
```
Comment (yujuhong) [inline test/e2e_node/runner/run_e2e.go]: typo: s/etcdCominedOutput/etcdCombinedOutput

File: test/e2e_node/runner/run_e2e.go (hunk 1 above)
Comment (yujuhong) [inline test/e2e_node/runner/run_e2e.go]: Could we upload it to google storage as other jenkins builds? 
 1. Our logs tend to be very chatty for the purpose of debugging. 
 2. We remove the google storage buckets less often, so the data is more durable.

Comment (yujuhong) [inline test/e2e_node/runner/run_e2e.go]: If we are going to do that in the future, just ignore my comment :-)

**Comment by vishh** [comment]
Other than fixing @yujuhong's comments, this PR LGTM

File: test/e2e_node/runner/run_e2e.go (hunk 2 above)
Comment (yujuhong) [inline test/e2e_node/runner/run_e2e.go]: same for the next two lines.

File: test/e2e_node/runner/run_e2e.go [hunk 3]
```diff
@@ -123,34 +148,57 @@ func WaitForUser() {
...
+		result.err = fmt.Errorf("Host %s failed to run command %v", host, err)
+		return result
 	}
 
 	kubeletBin := filepath.Join(kubeRoot, *kubeOutputRelPath, "kubelet")
 	kh, err := c.RunAndWaitTillHealthy(
 		true, true, "10255", healthyTimeoutDuration, "healthz", kubeletBin, "--api-servers", "http://127.0.0.1:8080",
-		"--logtostderr", "--address", "0.0.0.0", "--port", "10250")
-	defer func() { kh.TearDown() }()
+		"--v", "2", "--alsologtostderr", "--address", "0.0.0.0", "--port", "10250")
```
Comment (yujuhong) [inline test/e2e_node/runner/run_e2e.go]: `--v=4` is recommended for debugging kubelet issues, but the log tends to be long and will be better uploaded somewhere.

**Comment by pwittrock** [comment]
I added the /var/log/messages is pretty huge, and probably don't want to include it.  Where should we upload?

**Comment by yujuhong** [comment]
> I added the /var/log/messages is pretty huge, and probably don't want to include it. Where should we upload?

Can we leverage the existing test infra and just upload to GCS?
Perhaps reusing the script as well: https://github.com/kubernetes/kubernetes/blob/master/hack/jenkins/upload-to-gcs.sh

File: test/e2e_node/runner/run_e2e.go (hunk 2 above)
Comment (pwittrock) [inline test/e2e_node/runner/run_e2e.go]: Done

File: test/e2e_node/runner/run_e2e.go (hunk 3 above)
Comment (pwittrock) [inline test/e2e_node/runner/run_e2e.go]: SGTM.  Added a TODO for that.

**Comment by pwittrock** [comment]
Added a TODO to handle the /var/log/messages.

**Comment by pwittrock** [comment]
PTAL

**Comment by yujuhong** [comment]
Need a rebase :)

**Comment by pwittrock** [comment]
Thanks Yuju, not sure what causes those rebase issues.

**Pull request merged**